### ISSUE TITLE
[no-shadow] False positive with enum

### ISSUE THREAD

**Issue by danielleallopez** [typescript-eslint/typescript-eslint#3329]
We have created a very simple Typescript file, that contains two enums and two constants. They are exported and used somewhere else.

To our surprise, linter is throwing a no-shadow error. The enums are not defined anywhere else in the code, 100% guaranteed.

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "import/extensions": "off",
    "import/no-named-as-default": "off",
    "import/no-named-as-default-member": "off",
    "import/namespace": "off",
    "import/default": "off",
    "import/prefer-default-export": "off"
  }
}
```

```TS
export enum ComparisonOperator {
    EQ,
    NEQ,
}

export enum RelationOperator {
    AND = 'AND',
    OR = 'OR',
    XOR = 'XOR',
}
```

**Expected Result**

No lint error since enum has been defined just once.

**Actual Result**

Error: 'ComparisonOperator' is already declared in the upper scope on line 1 column 13. eslintno-shadow

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.22.0` |
| `@typescript-eslint/parser`        | `4.22.0` |
| `TypeScript`                       | `4.2.4` |
| `ESLint`                           | `7.24.0` |
| `node`                             | `14.14.0` |

**Comment by bradzacher** [comment]
> - [x] I have read the FAQ and my problem is not listed.

Did you though?
https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code
